src/SourceCode.Chasm.Repository.Disk/DiskChasmRepo.cs

```diff
@@ -23,15 +23,24 @@ public sealed partial class DiskChasmRepo : ChasmRepository
 
         #endregion
 
+        #region Properties
+
+        /// <summary>
+        /// Gets the root path for the repository.
+        /// </summary>
+        public string RootPath { get; }
+
+        #endregion
+
         #region Constructors
 
         public DiskChasmRepo(string rootFolder, IChasmSerializer serializer, CompressionLevel compressionLevel, int maxDop)
             : base(serializer, compressionLevel, maxDop)
         {
             if (string.IsNullOrWhiteSpace(rootFolder) || rootFolder.Length <= 2) throw new ArgumentNullException(nameof(rootFolder)); // "C:\" is shortest permitted path
-            if (rootFolder[rootFolder.Length - 1] != Path.DirectorySeparatorChar) throw new ArgumentException("Path must end with " + Path.DirectorySeparatorChar, nameof(rootFolder));
```

Comment:
Why did you remove the guards

Reply:
They didn't prevent any exceptions from occurring later. Everything uses `Path.Combine`.